File: java2csharp/java/Samsung/Look.EdgeScreen/list/DBHelper.java

package com.example.edgescreen.list;

import java.util.ArrayList;

public class DBHelper {
    private ArrayList<String> mDatalist = new ArrayList<String>();

    private static DBHelper sInstance = new DBHelper();

    private Object mLock = new Object();

    public static DBHelper getInstance() {
        return sInstance;
    }

    public void addData(String data) {
        synchronized (mLock) {
            mDatalist.add(data);
        }
    }

    public String getData(int index) {
        synchronized (mLock) {
            try {
                return mDatalist.get(index);
            } catch (IndexOutOfBoundsException e) {
                return null;
            }
        }
    }

    public int getSize() {
        synchronized (mLock) {
            return mDatalist.size();
        }
    }

    public void clearData() {
        synchronized (mLock) {
            mDatalist.clear();
        }
    }
}

File: java2csharp/csharp/Samsung/Look.EdgeScreen/list/DBHelper.cs
using System.Collections.Generic;

namespace com.example.edgescreen.list
{

	public class DBHelper
	{
		private List<string> mDatalist = new List<string>();

		private static DBHelper sInstance = new DBHelper();

		private object mLock = new object();

		public static DBHelper Instance
		{
			get
			{
				return sInstance;
			}
		}

		public virtual void addData(string data)
		{
			lock (mLock)
			{
				mDatalist.Add(data);
			}
		}

		public virtual string getData(int index)
		{
			lock (mLock)
			{
				try
				{
					return mDatalist[index];
				}
				catch (System.IndexOutOfRangeException)
				{
					return null;
				}
			}
		}

		public virtual int Size
		{
			get
			{
				lock (mLock)
				{
					return mDatalist.Count;
				}
			}
		}

		public virtual void clearData()
		{
			lock (mLock)
			{
				mDatalist.Clear();
			}
		}
	}

}